### PR TITLE
Prevent /v1/enumerate from returning frameworks: null, or tasks: null

### PR DESCRIPTION
**Pull request by sargun**

### CODE REVIEW

File: records/generator.go [hunk 1]
```diff
@@ -436,7 +442,10 @@ func (rg *RecordGenerator) listenerRecord(listener string, ns string) {
 
 func (rg *RecordGenerator) taskRecords(sj state.State, domain string, spec labels.Func, ipSources []string) {
 	for _, f := range sj.Frameworks {
-		enumerableFramework := &EnumerableFramework{Name: f.Name}
+		enumerableFramework := &EnumerableFramework{
+			Name: f.Name,
+			Tasks: []*EnumerableTask{},
```
Comment (jdef) [inline records/generator.go]: can't you just use a json `omitempty` tag for `Tasks` in the struct where it's defined?

File: records/generator.go [hunk 2]
```diff
@@ -111,7 +111,13 @@ type EnumerationData struct {
 
 // NewRecordGenerator returns a RecordGenerator that's been configured with a timeout.
 func NewRecordGenerator(httpTimeout time.Duration) *RecordGenerator {
-	rg := &RecordGenerator{httpClient: http.Client{Timeout: httpTimeout}}
+	enumData := EnumerationData{
+		Frameworks: []*EnumerableFramework{},
```
Comment (jdef) [inline records/generator.go]: can't you just use a json `omitempty` tag for `Frameworks` in the struct where it's defined?

File: records/generator.go (hunk 1 above)
Comment (sargun) [inline records/generator.go]: We don't want to entirely omit it, we want it to be [] when there are no tasks.

File: records/generator.go (hunk 2 above)
Comment (sargun) [inline records/generator.go]: Nope, it should be [] on no frameworks, not null or missing entirely

**Comment by jdef** [comment]
lgtm

**Pull request merged**